src/komodo_notary.cpp

```diff
@@ -283,92 +283,80 @@ int32_t komodo_chosennotary(int32_t *notaryidp,int32_t height,uint8_t *pubkey33,
     return(modval);
 }
 
-//struct komodo_state *komodo_stateptr(char *symbol,char *dest);
-
-struct notarized_checkpoint *komodo_npptr_for_height(int32_t height, int *idx)
+/******
+ * @brief Search the notarized checkpoints for a particular height
+ * @note Finding a mach does include other criteria other than height
+ *      such that the checkpoint includes the desired hight
+ * @param height the key
+ * @returns the checkpoint or nullptr
+ */
+const notarized_checkpoint *komodo_npptr(int32_t height)
 {
-    char symbol[KOMODO_ASSETCHAIN_MAXLEN],dest[KOMODO_ASSETCHAIN_MAXLEN]; int32_t i; struct komodo_state *sp; struct notarized_checkpoint *np = 0;
-    if ( (sp= komodo_stateptr(symbol,dest)) != 0 )
+    char symbol[KOMODO_ASSETCHAIN_MAXLEN];
+    char dest[KOMODO_ASSETCHAIN_MAXLEN]; 
+    komodo_state *sp;
+
+    if ( (sp= komodo_stateptr(symbol,dest)) != nullptr )
     {
-        for (i=sp->NUM_NPOINTS-1; i>=0; i--)
-        {
-            *idx = i;
-            np = &sp->NPOINTS[i];
-            if ( np->MoMdepth != 0 && height > np->notarized_height-(np->MoMdepth&0xffff) && height <= np->notarized_height )
-                return(np);
-        }
+        return sp->CheckpointAtHeight(height);
     }
-    *idx = -1;
-    return(0);
-}
-
-struct notarized_checkpoint *komodo_npptr(int32_t height)
-{
-    int idx;
-    return komodo_npptr_for_height(height, &idx);
-}
-
-struct notarized_checkpoint *komodo_npptr_at(int idx)
-{
-    char symbol[KOMODO_ASSETCHAIN_MAXLEN],dest[KOMODO_ASSETCHAIN_MAXLEN]; struct komodo_state *sp;
-    if ( (sp= komodo_stateptr(symbol,dest)) != 0 )
-        if (idx < sp->NUM_NPOINTS)
-            return &sp->NPOINTS[idx];
-    return(0);
+    return nullptr;
 }
 
+/****
+ * Search for the last (chronological) MoM notarized height
+ * @returns the last notarized height that has a MoM
+ */
 int32_t komodo_prevMoMheight()
 {
-    static uint256 zero;
-    char symbol[KOMODO_ASSETCHAIN_MAXLEN],dest[KOMODO_ASSETCHAIN_MAXLEN]; int32_t i; struct komodo_state *sp; struct notarized_checkpoint *np = 0;
-    if ( (sp= komodo_stateptr(symbol,dest)) != 0 )
+    char symbol[KOMODO_ASSETCHAIN_MAXLEN];
+    char dest[KOMODO_ASSETCHAIN_MAXLEN];
+    komodo_state *sp;
+
+    if ( (sp= komodo_stateptr(symbol,dest)) != nullptr )
```

Comment:
```suggestion
    komodo_state *sp = komodo_stateptr(symbol,dest);
    if ( sp != nullptr )
```
Just to get code more clear.